detectable/src/main/java/com/synopsys/integration/detectable/detectables/go/gomod/GoModDataGatherer.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * detectable
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Use subject to the terms and conditions of the Synopsys End User Software License and Maintenance Agreement. All rights reserved worldwide.
+ */
+package com.synopsys.integration.detectable.detectables.go.gomod;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import com.synopsys.integration.detectable.ExecutableTarget;
+import com.synopsys.integration.detectable.detectable.exception.DetectableException;
+import com.synopsys.integration.detectable.detectables.go.gomod.model.GoGraphRelationship;
+import com.synopsys.integration.detectable.detectables.go.gomod.model.GoListAllData;
+import com.synopsys.integration.detectable.detectables.go.gomod.model.GoListModule;
+import com.synopsys.integration.detectable.detectables.go.gomod.parse.GoGraphParser;
+import com.synopsys.integration.detectable.detectables.go.gomod.parse.GoListParser;
+import com.synopsys.integration.detectable.detectables.go.gomod.parse.GoModWhyParser;
+import com.synopsys.integration.executable.ExecutableRunnerException;
+
+public class GoModDataGatherer {
```

Comment:
How do we feel about a helper class like this for the Extractor. It's responsible for gathering the data. The extractor would be the glue for data gathering and processing.

Reply:
Would you test the DataGatherer? If the answer is yes, I could see it. If the answer is no, I'd argue it should be in the extractor. I don't necessarily see immediately what value it provides, it looks like non-tested glue code I expect in an extractor (and in this case by glue I mean joining executable output and the code that parses it).

Reply:
It's not tested outside of the Detectable tests, same as the Extractor. Eric had mentioned that area was noisy and would like it in another class. It is the same un-tested glue code, just separated for readability. Perhaps another method in the Extractor would be better than a new class.

Reply:
Yeah then I find it a better home in the extractor, though we do have some noisy extractors so we may need to do something. At least for those even breaking them up into the methods might reduce the noise enough. 
```
List<GoListModile> listModules() { 
        List<String> listOutput = goModCommandExecutor.generateGoListOutput(directory, goExe);
        return goListParser.parseGoListModuleJsonOutput(listOutput); 
}
```